src/video/SDL_video.c

```diff
@@ -1256,21 +1256,22 @@ SDL_DisplayID SDL_GetDisplayForWindow(SDL_Window *window)
      * (for example if the window is off-screen), but other code may expect it
      * to succeed in that situation, so we fall back to a generic position-
      * based implementation in that case. */
-    if (!displayID && (window->flags & SDL_WINDOW_FULLSCREEN)) {
-        displayID = window->fullscreen_mode.displayID;
-    }
     if (!displayID) {
-        displayID = SDL_GetDisplayForWindowCoordinate(window->windowed.x);
```

Comment:
This is actually supposed to use window->windowed.x, because the user can provide CENTERED or UNDEFINED with a display ID for these values.